audio/drivers/coreaudio.cpp

```diff
@@ -0,0 +1,409 @@
+    desiredFormat.mBytesPerFrame = sizeof(float) * 2;
+    desiredFormat.mChannelsPerFrame = 2;
+    status = AudioUnitSetProperty(*audioUnit, kAudioUnitProperty_StreamFormat, kAudioUnitScope_Input, 0, &desiredFormat, sizeof(desiredFormat));
+    if(status) {
+        qDebug("Failed to set stream format");
+        goto error;
+    }
+    
+    AURenderCallbackStruct renderCallbackData;
+    renderCallbackData.inputProc = CoreAudioIOProc;
+    renderCallbackData.inputProcRefCon = ca;
+    
+    status = AudioUnitSetProperty(*audioUnit, kAudioUnitProperty_SetRenderCallback, kAudioUnitScope_Output, 0, &renderCallbackData, sizeof(renderCallbackData));
+    if(status) {
+        qDebug("Failed to set render callback");
+        goto error;
+    }
+    
+    AudioUnitInitialize(*audioUnit);
+    
+    return true;
+   error:
+    if(*audioUnit) {
+        AudioComponentInstanceDispose(*audioUnit);
+        *audioUnit = NULL;
+    }
+    *audioDevice = kAudioDeviceUnknown;
+    return false;
+}
+
+static OSStatus CoreAudioIOProc( void *inRefCon,
+                               AudioUnitRenderActionFlags *ioActionFlags,
+                               const AudioTimeStamp *inTimeStamp,
+                               UInt32 inBusNumber,
+                               UInt32 inNumberFrames,
+                               AudioBufferList *ioData)
+{
+      CoreAudio* ca = static_cast<CoreAudio*>(inRefCon);
+
+      seq->setInitialMillisecondTimestampWithLatency();
+
+      long bytesPerFrame = sizeof(float) * ioData->mBuffers[0].mNumberChannels;
+
+      float* buffer = static_cast<float*>(ioData->mBuffers[0].mData);
+      UInt64 frames = ioData->mBuffers[0].mDataByteSize / bytesPerFrame;
+
+      seq->process((unsigned)frames, buffer);
+
+      return noErr;
+}
+
+bool CoreAudio::init(bool)
+      {
+	SInt32 major;
+	SInt32 minor;
+	Gestalt(gestaltSystemVersionMajor, &major);
+	Gestalt(gestaltSystemVersionMinor, &minor);
+	
+      initialized = true;
+    
+	// Starting with 10.6 systems, the HAL notification thread is created internally
+	if (major == 10 && minor >= 6) {
+		CFRunLoopRef theRunLoop = NULL;
+		AudioObjectPropertyAddress theAddress = { kAudioHardwarePropertyRunLoop, kAudioObjectPropertyScopeGlobal, kAudioObjectPropertyElementMaster };
+		OSStatus osErr = AudioObjectSetPropertyData (kAudioObjectSystemObject, &theAddress, 0, NULL, sizeof(CFRunLoopRef), &theRunLoop);
+		if (osErr != noErr) {
+                  initialized = false;
+                  return false;
+		}
+	}
+  
+      AudioObjectPropertyAddress outputDeviceAddress = {
+            kAudioHardwarePropertyDefaultOutputDevice,
+            kAudioObjectPropertyScopeGlobal,
+            kAudioObjectPropertyElementMaster
+      };
+
+      AudioObjectAddPropertyListener(kAudioObjectSystemObject, &outputDeviceAddress, &onDefaultOutputDeviceChanged, this);
+
+      AudioDeviceID deviceId = getDefaultOutputDevice();
+      return openAudioUnit(&audioUnit, &deviceID);
```

Comment:
How does this compile on your machine and on Travis CI? The spelling of `deviceID` does not match the way it is spelled in the above line, and you forgot to pass `this` as the third parameter to `openAudioUnit`.

Reply:
It doesn't. I made the mistake of refactoring some unneeded code out of the file, and not compiling after I refactored. Apologies for that oversight.

Reply:
I will address this in a commit today.

Reply:
Fixed in 8d8ed08eb